include/macroni/Translation/MacroniCodeGenVisitorMixin.hpp

```diff
@@ -101,8 +103,24 @@ struct MacroniCodeGenVisitorMixin
     }
 
     // Get the substitution's location, name, parameter names, and whether it is
-    // function-like
-    auto loc = meta_location(stmt);
+    // function-like.
+    //
+    // TODO(bpp): Call meta_location instead of directly setting the location
+    // here like this. We can't right now because codegen uses OOP inheritance
+    // for the metagen and the base class doesn't have a location method for
+    // macro substitutions.
```

Comment:
Move this to your class anyway, and here do normal C++ dynamic cast to get your inherited type. If it fails something is bonkers anyway and I would just emit a hard error, but also unknown loc can be set instead.